bindings/experimental/distrdf/python/DistRDF/Ranges.py

```diff
@@ -111,6 +117,7 @@ def get_clusters(treename, filelist):
     return clusters
 
 
+@lru_cache(maxsize=None)
```

Comment:
So this caches the result of the function given the arguments, right?
Should we set a boundary in the number of distinct calls that are cached, just in case?

The possible problem with this is if the content of the input files changes during the execution (which I guess is unlikely).

Reply:
> So this caches the result of the function given the arguments, right?

Correct, and the arguments all need to be hashable (that's why where we previously had lists now there are tuples)

> Should we set a boundary in the number of distinct calls that are cached, just in case?

It can be done sure. I just don't know what's the magic number ahah

> The possible problem with this is if the content of the input files changes during the execution (which I guess is unlikely).

This is actually something I wanted to explore. In theory we could work around this if the TFile fUUID member does what I think it does, i.e. associates a unique id to the same filename each time it gets changed.

Reply:
I would just leave the default for the maxsize.

Reply:
I agree with Enric, it's weird that we are saying "we are ok with using an unbounded amount of memory"